### PR TITLE
'Type' field added to Salary Component

### PR DESCRIPTION
**Pull request by KanchanChauhan**

### CODE REVIEW

File: erpnext/patches/v7_1/update_component_type.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import frappe
+from frappe.utils import flt
+
+def execute():
```
Comment (nabinhait) [inline erpnext/patches/v7_1/update_component_type.py]: This patch can be written in one line:

frappe.db.sql("""update `tabSalary Component` set type=if(parentfield='earnings', 'Earning', 'Deduction')""")

File: erpnext/hr/doctype/salary_structure/salary_structure.js [hunk 2]
```diff
@@ -13,6 +13,23 @@ cur_frm.cscript.onload = function(doc, dt, dn){
 }
 
 frappe.ui.form.on('Salary Structure', {
+	onload: function(frm) {
+		frm.set_query("salary_component", "earnings", function() {
+			return {
+				filters: {
+					type: "earning"
```
Comment (nabinhait) [inline erpnext/hr/doctype/salary_structure/salary_structure.js]: type is case sensitive?

File: erpnext/patches/v7_1/update_component_type.py (hunk 1 above)
Comment (KanchanChauhan) [inline erpnext/patches/v7_1/update_component_type.py]: 'parentfield' is in Salary Detail actually.

File: erpnext/hr/doctype/salary_structure/salary_structure.js (hunk 2 above)
Comment (KanchanChauhan) [inline erpnext/hr/doctype/salary_structure/salary_structure.js]: No

File: erpnext/patches/v7_1/update_component_type.py (hunk 1 above)
Comment (nabinhait) [inline erpnext/patches/v7_1/update_component_type.py]: Sorry I missed that.

**Pull request merged**